### DISCUSSION TITLE
Just solve linear equations without objective function

### DISCUSSION THREAD

**Discussion by ravikp** [google/or-tools#2533] [Linear Solver questions] [answered]
Hello,

I started learning about google or-tools through the [available documentation](https://developers.google.com/optimization). In the initial examples, there is always an objective function. Is it possible to solve a bunch of linear equations without an objective function?

**Comment by lperron** [comment]
which solver ?

**Reply by ravikp** [reply]
I could solve this problem 

`3x + y = 13`
and 
`2x - y = 2`

using CP-SAT solver. But if the x, y variables are rational numbers and not just limited to integers. How do I solve them, without the objective function?

**Comment by lperron** [comment, marked as answer]
use the linear solver.
Using an objective is not mandatory. The solvers are less efficient without
one.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by ravikp** [reply]
To solve

`3x + y = 14.5`
and 
`2x - y = 3`

I used the following code, let me know if this is the right approach

```python
from ortools.linear_solver import pywraplp

solver = pywraplp.Solver.CreateSolver('GLOP')

x = solver.NumVar(0, solver.infinity(), 'x')
y = solver.NumVar(0, solver.infinity(), 'y')

ct1 = solver.Constraint(14.5, 14.5, 'ct')
ct1.SetCoefficient(x, 3)
ct1.SetCoefficient(y, 1)

ct2 = solver.Constraint(3, 3, 'ct')
ct2.SetCoefficient(x, 2)
ct2.SetCoefficient(y, -1)

# Create the blank objective function
objective = solver.Objective()

## invoke the solver and display results

solver.Solve()
print('x = ', x.solution_value())
print('y = ', y.solution_value())

```

**Reply by ravikp** [reply]
I got the answer, but I wanted to confirm with you the approach of solving the equations.

**Reply by Mizux** [reply]
Yes it's OK but you can simply comment out the dummy `objective = solver.Objective()`.

Since you are using python, you could also implement it using the natural API...
```python
from ortools.linear_solver import pywraplp

solver = pywraplp.Solver.CreateSolver('GLOP')

x = solver.NumVar(0, solver.infinity(), 'x')
y = solver.NumVar(0, solver.infinity(), 'y')

solver.Add(3 * x + y == 14.5)
solver.Add(2 * x - y == 3)

## invoke the solver and display results
solver.Solve()
print('x = ', x.solution_value())
print('y = ', y.solution_value())
```

**Reply by ravikp** [reply]
Thanks a lot @Mizux

**Comment by ravikp** [comment]
@lperron - We can close this ticket. Thanks a lot.
